flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/HasMaxIter.java

```diff
@@ -26,7 +26,7 @@
 /** Interface for the shared maxIter param. */
 public interface HasMaxIter<T> extends WithParams<T> {
     Param<Integer> MAX_ITER =
-            new IntParam("maxIter", "Maximum number of iterations.", 20, ParamValidators.gtEq(0));
+            new IntParam("maxIter", "Maximum number of iterations.", 20, ParamValidators.gt(0));
```

Comment:
Could we just update all algorithms to support `maxIter=0` instead of disallowing it?

Reply:
I prefer not. Is there a real use case for training a model iteratively while setting `MaxIterationNumber` to zero?

Reply:
OK. Then let's keep it as is.